apps/system/js/quick_settings.js

```diff
@@ -170,32 +170,19 @@
      * @memberof QuickSettings.prototype
      */
     monitorBluetoothChange: function() {
-      var self = this;
-      var btFirstSet = true;
-      SettingsListener.observe('bluetooth.enabled', true, function(value) {
-        // check self.bluetooth.dataset.enabled and value are identical
-        if ((self.bluetooth.dataset.enabled && value) ||
-          (self.bluetooth.dataset.enabled === undefined && !value)) {
-          return;
-        }
-
+      var req = navigator.mozSettings.createLock().get('bluetooth.enabled');
```

Comment:
Don't think we need to get the settings here. Use Service.query('Bluetooth.isEnabled') and delete <code>this.bluetooth.dataset.initializing</code> only when the value is not undefined - because the remaining event part is also relying on bluetooth(v1/v2).js to tell you it's enabled or not.